fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/Compensation.java

```diff
@@ -739,6 +747,7 @@ public RelationalExpression apply(@Nonnull final Memoizer memoizer, @Nonnull Rel
 
             final var matchedAliases =
                     matchedQuantifierMap.keySet();
+            Verify.verify(compensatedAliases.equals(matchedAliases));
```

Comment:
(writing comment here, since I can't add a comment on unchanged code).

In lin 740, we short-circuit the calculation if the predicate compensation map is empty, do we need to rework this to handle potential for-each unmatched quantifiers that could affect the cardinality _now_? or this can wait until we have proper support for join indexes?